universum/modules/code_report_collector.py

```diff
@@ -73,7 +76,17 @@ def _report_as_sarif_json(self, report) -> int:
                         if location_data.get('address'):
                             continue  # binary artifact can't be processed
                         raise ValueError("Unexpected lack of artifactLocation tag")
-                    path: str = urllib.parse.unquote(artifact_data.get('uri', ''))
+                    if artifact_data.get('uriBaseId'):
+                        uri = artifact_data.get('uri')
+                        if not uri:
+                            raise ValueError("Unexpected lack of uri tag")
+                        uri_base_id = artifact_data.get('uriBaseId', '')
+                        root_base_path: str = urllib.parse.urlparse(root_uri_base_ids.get(uri_base_id)).path
```

Comment:
Please move url parsing to outside of two for loops

Reply:
URL parsing moved out of the loops